scheduler/src/cook/mesos/util.clj

```diff
@@ -267,9 +267,22 @@
 
 (defn same-user-task-comparator
   "Comparator to order same user's tasks"
-  []
-  (fn [task1 task2]
-      (compare (task->feature-vector task1) (task->feature-vector task2))))
+  ([]
+   (same-user-task-comparator []))
+  ([tasks]
+   ;; Pre-compute the feature-vector for tasks we expect to see to improve performance
+   ;; This is done because accessing fields in datomic entities is much slower than
+   ;; a map access, even when accessing multiple times. 
+   ;; Don't want to complicate the function by caching new values in the event we see them
+   (let [task-ent->feature-vector (into {} 
```

Comment:
Yes, good point. Pushed a new version, it is much better thanks to your suggestion